pkg/pillar/pubsub/updaters.go

```diff
@@ -30,7 +30,8 @@ func (u *Updaters) Add(updater chan Notify, name string, instance int) {
 // Remove an updater
 func (u *Updaters) Remove(updater chan Notify) {
 	u.lock.Lock()
-	servers := make([]notifyName, len(u.servers))
+	// A new slice which points to the same underlying array
+	servers := u.servers[:0]
```

Comment:
Sure, it works. But how much memory did it really use?

Reply:
Since it doubles each time, it can be a lot. On @naiming-zededa s device nim ended up with a 30Mbyte array here.
Note that the intent was to set the capacity and not the size, but a '0' argument is missing in the original code. 
But while touching it we might as well make it not reallocate the array when it shrinks. Max size is the number of subscribers which is still less than 10 for the most popular topics, so little benefit of making it reallocate a smaller array when it shrinks.

Reply:
Heh, well that might be significant! I am surprised it would grow unendingly. At the end of each routine, would it not garbage collect the slice?

Reply:
The slice grows since it starts with N entries (which are empty) from the make() and then we append N-1 entries (all but the one which is being removed). Hence len keeps basically doubling for each Remove call.
Note that the routine which walks the server list silently skips entries which doesn't match a Name, which includes those uninitialized slots from the make().